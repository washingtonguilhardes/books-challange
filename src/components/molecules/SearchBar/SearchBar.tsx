import React from 'react';

import AppBar from '@material-ui/core/AppBar';
import InputBase from '@material-ui/core/InputBase';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import SearchIcon from '@material-ui/icons/Search';

import { Button } from '@src/components/athoms';

import { useSearchBarStyles } from './SearchBar.styles';
import { SearchBarProps } from './SearchBar.types';

export function SearchBar(props: SearchBarProps) {
  const { onSubmit, title } = props;
  const classes = useSearchBarStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.title} variant="h6" noWrap>
          {title}
        </Typography>
        <div className={classes.grow} />
        <div className={classes.search}>
          <div className={classes.searchIcon}>
            <SearchIcon />
          </div>
          <InputBase
            placeholder="Searchâ€¦"
            onChange={ev => onSubmit(ev.target.value)}
            classes={{
              root: classes.inputRoot,
              input: classes.inputInput,
            }}
            inputProps={{ 'aria-label': 'search' }}
          />
          <Button variant="text" textColor="#fff">
            SEARCH
          </Button>
        </div>
      </Toolbar>
    </AppBar>
  );
}
