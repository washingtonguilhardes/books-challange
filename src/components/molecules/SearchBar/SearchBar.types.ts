import React from 'react';

import { TextFieldProps } from '@material-ui/core';

export interface SearchBarProps {
  title?: React.ReactNode;
  TextFieldProps?: TextFieldProps;
  onSubmit: (_text: string) => void;
}
