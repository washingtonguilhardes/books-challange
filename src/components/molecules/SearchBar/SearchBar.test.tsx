import React from 'react';

import { render } from '@testing-library/react';

import { SearchBar } from './SearchBar';

describe('SearchBar', () => {
  it('should match with snapshot', () => {
    const onSubmitFn = jest.fn();
    const { container } = render(<SearchBar onSubmit={onSubmitFn} title="title" />);
    expect(container).toMatchSnapshot();
  });
});
