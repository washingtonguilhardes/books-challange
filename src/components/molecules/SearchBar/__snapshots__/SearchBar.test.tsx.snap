// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match with snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-2 MuiTypography-h6 MuiTypography-noWrap"
      >
        title
      </h6>
      <div
        class="makeStyles-grow-1"
      />
      <div
        class="makeStyles-search-3"
      >
        <div
          class="makeStyles-searchIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          class="MuiInputBase-root makeStyles-inputRoot-5"
        >
          <input
            aria-label="search"
            class="MuiInputBase-input makeStyles-inputInput-6"
            placeholder="Searchâ€¦"
            type="text"
            value=""
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-7 makeStyles-root-8"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            SEARCH
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
